shared/js/background/settings.js

```diff
@@ -124,6 +126,9 @@ function updateSetting (name, value) {
 
     settings[name] = value
     syncSettingTolocalStorage()
+    setTimeout(() => {
```

Comment:
What's the timeout for?

Reply:
Since the events are dispatched synchronously otherwise. I figure that if someone updates a setting and waits for the update to finish, it would be undesirable to also wait for all the update listeners to finish. I don't think it's too important for now, but I wanted to avoid a foot-gun for the next person using this.

Reply:
This won't guarantee it's stored though, so you might end up creating more problems.

I would suggest making `syncSettingTolocalStorage` return the promise and await/then it here if you want to provide any async guarantee.

Reply:
Good idea, Done.